src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -313,7 +314,7 @@ private TransactionReceipt MakeConfirmTransaction(UInt256 cycle, ThresholdKeyrin
                 keyring.TpkePublicKey.ToBytes(),
                 keyring.ThresholdSignaturePublicKeySet.Keys.Select(key => key.ToBytes()).ToArray()
             );
-            return _transactionSigner.Sign(tx, _privateWallet.EcdsaKeyPair);
+            return _transactionSigner.Sign(tx, _privateWallet.EcdsaKeyPair, HardforkHeights.IsHardfork_6Active(_blockManager.GetHeight()));
```

Comment:
it should be _blockManager.GetHeight() + 1, 
or we can pass the blockHeight while calling.